### PR TITLE
[Dev to staging]

### PR DESCRIPTION
**Pull request by sofiiagran**
## Description

This is a dev to staging PR, this version has a lot of linting fixes as well.

Addressing the problem where only exercise components inside the swiper had isComplete set to true has been resolved. Now, all components are updated, and the swiper starts at the first incomplete component, if any, instead of the start. Additionally, there is a refactoring of the points logic, but the underlying logic remains unchanged.

## Changes

The changes are related to issue from mobile https://github.com/Educado-App/educado-mobile/issues/199

* Commented out some stuff:
       * commented out leaderboard.js, since it is not used and gives linting errors
       * commented out function ensureNewValues in validation.js since it gives linting error and is not in use

## Related Issues

Issue https://github.com/Educado-App/educado-mobile/issues/199

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Notes for Reviewers

Not a biggie, mostly linting

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -21,22 +22,13 @@ module.exports = {
 	'parserOptions': {
 		'ecmaVersion': 'latest'
 	},
+	'ignorePatterns': [
+		'**/*.test.js', '**/*.spec.js', '**/*.config.js', 'jest-setup.js', 'db.js'
```
Comment (MaizeHH) [inline .eslintrc.js:26]: why are these ignored?

**Review by Mafusn — APPROVED** [review]
Looks amazing.

File: helpers/patterns.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 const regexPatterns = {
-	email: /^[a-z0-9!'#$%&*+\/=?^_`{|}~-]+(?:\.[a-z0-9!'#$%&*+\/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-zA-Z]{2,}$/i,
```
Comment (MaizeHH) [inline helpers/patterns.js:2]: these slashes after the escaping character should be reinserted

File: routes/studentRoutes.js [hunk 3]
```diff
@@ -1,241 +1,220 @@
...
-  }
-
-  const id = mongoose.Types.ObjectId(req.params.id);
-
-  const { points, level } = req.body;
-
-  let student = await StudentModel.findOne({ baseUser: id });
-
-  let updatedUser;
+router.get('/:id/info', async (req, res) => {
```
Comment (MaizeHH) [inline routes/studentRoutes.js:11]: the 'info' should be removed from this route

**Review by MaizeHH — CHANGES_REQUESTED** [review]
Looks good! However I have a few requested changes 🌵

File: helpers/patterns.js (hunk 2 above)
Comment (sofiiagran) [reply helpers/patterns.js:2]: @MaizeHH I'm pretty sure this is your groups fixes. I sent you like a month ago some linting errors I got and you fixed it and pushed. However, it was a mistake so they were pushed to staging immidately, and therefore, someone overwrote the changes. So this is actually a copy from your group commit (I think Morten).

File: routes/studentRoutes.js (hunk 3 above)
Comment (sofiiagran) [reply routes/studentRoutes.js:11]: @MaizeHH why?

File: helpers/patterns.js (hunk 2 above)
Comment (sofiiagran) [reply helpers/patterns.js:2]: see commit fc6eff1

Comment (MaizeHH) [reply helpers/patterns.js:2]: Ok 👍 we'll make a hotfix for it

File: routes/studentRoutes.js (hunk 3 above)
Comment (MaizeHH) [reply routes/studentRoutes.js:11]: the api guideline does it like that. we'll have a look and see if it is worth fixing at this point

Comment (Donhusum) [reply routes/studentRoutes.js:11]: This was also mentioned to the group who made this. If you change it it should be changed to '/info/:id'
If you change it, it should also be changed on the app

**Pull request merged**